packages/compass-connections/src/hooks/use-connection-repository.ts

```diff
@@ -67,7 +81,16 @@ export function useConnectionRepository(): ConnectionRepository {
     ConnectionInfo | undefined
   >(undefined);
 
-  const persistOIDCTokens = usePreference('persistOIDCTokens');
+  const favoriteConnectionsRef = useRef(favoriteConnections);
+  favoriteConnectionsRef.current = favoriteConnections;
+
+  const nonFavoriteConnectionsRef = useRef(nonFavoriteConnections);
+  nonFavoriteConnectionsRef.current = nonFavoriteConnections;
+
+  const autoConnectInfoRef = useRef(autoConnectInfo);
+  autoConnectInfoRef.current = autoConnectInfo;
```

Comment:
Drive-by: in a lot of places we need to just access the current state of those, which without changing some of the methods here required subscribing to changes to those (invalidating dependencies of a loot of other hooks) even when we don't care about it and only want to read the current value